qiita_db/util.py

```diff
@@ -520,15 +521,53 @@ def get_files_from_uploads_folders(study_id):
         List of the filepaths for upload for that study
     """
     fp = []
-    for _, p in get_mountpoint("uploads", retrive_all=True):
+    for pid, p in get_mountpoint("uploads", retrive_all=True):
         t = join(p, study_id)
         if exists(t):
-            fp.extend([f for f in listdir(t) if not f.startswith('.') and
-                      not isdir(join(t, f))])
+            fp.extend([(pid, f) for f in listdir(t) if not f.startswith('.')
+                      and not isdir(join(t, f))])
 
     return fp
 
 
+def move_upload_files_to_trash(study_id, files_to_move):
+    """Retrive files in upload folders
+
+    Parameters
+    ----------
+    study_id : int
+        The study id of which to retrive all upload folders
+    files_to_move : list
+        List of tuples (folder_id, filename)
+
+    Raises
+    ------
+    QiitaDBError
+        If folder_id or the study folder don't exist
+    """
+    folders = {k: v for k, v in get_mountpoint("uploads", retrive_all=True)}
+
+    for fid, filename in files_to_move:
+        if fid not in folders:
+            raise QiitaDBError("Filepath not stored in the database")
+
+        foldername = join(folders[fid], str(study_id))
+        if not exists(foldername):
+            raise QiitaDBError("Filepath not stored in the database")
+
+        trashpath = join(foldername, 'trash')
```

Comment:
Conceptually this is a problem: the qiita_db code then requires GUI
validation.

On Tue, Feb 3, 2015 at 2:15 PM, Antonio Gonzalez <notifications@github.com>
wrote:

> In qiita_db/util.py
> <https://github.com/biocore/qiita/pull/815#discussion_r24040477>:
>
> > +    Raises
> > +    ------
> > +    QiitaDBError
> > +        If folder_id or the study folder don't exist
> > +    """
> > +    folders = {k: v for k, v in get_mountpoint("uploads", retrive_all=True)}
> > +
> > +    for fid, filename in files_to_move:
> > +        if fid not in folders:
> > +            raise QiitaDBError("Filepath not stored in the database")
> > +
> > +        foldername = join(folders[fid], str(study_id))
> > +        if not exists(foldername):
> > +            raise QiitaDBError("Filepath not stored in the database")
> > +
> > +        trashpath = join(foldername, 'trash')
>
> @wasade <https://github.com/wasade> You shouldn't be able to upload a
> file named trash as the GUI doesn't allow it as it needs a valid extension
> to be uploaded. Thanks for the clarification. In your opinion, is there
> something else missing in this PR?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/815/files#r24040477>.
>